### DISCUSSION TITLE
Raspberry Pi 1 Armv6 Support?

### DISCUSSION THREAD

**Discussion by azzajess** [grafolean/grafolean#32] [Q&A]
Hey, im trying to get this working on a raspberry Pi 1 and I get error messages like
`mosquitto The requested image's platform (linux/arm/v7) does not match the detected host platform (linux/arm/v6) and no specific platform was requested`
This happens for mosquitto, pingbot netflowwriter, snmpbot, netflowbot, netflowcollector and grafolean 
Then at the end of `docker compose up -d` i get `dependency failed to start: container for service "db" is unhealthy`

Im wondering if this is or can be compatible with armv6?

Thanks

**Comment by grafolean** [comment]
Hi, I'm not sure - I would need to check if I can get this to build for armv6... Sounds interesting though. It might take a day or two to find the time to do it, but I'll look into it.

**Reply by grafolean** [reply]
Grafolean uses Debian as its base for the Docker images, but it looks like Debian does not support arm32v6, only [arm32v5](https://github.com/docker-library/python/issues/481). With ARMv5 python image seems to [exist](https://hub.docker.com/r/arm32v5/python), but node.js does not, at least I could not find it. I have tried to build it on top of [debian image](https://hub.docker.com/r/arm32v5/debian/) directly, but it turns out that `nodejs` package is not available for this architecture.

The way I see it, the options are:
- finding a way to install node.js on debian/armv5, or
- replacing the installation procedure to use alpine images instead of debian (which sounds like a lot of work).

Both of these would be difficult for me to do since I don't have access to a RaspberryPi 1. If you are motivated to try it yourself, however, I would be happy to help if you get stuck.

Some hints: Grafolean is built using only a single Dockerfile which is in `install/docker-compose/Dockerfile`. The Docker images which are available on Docker Hub are build using the GitHub Actions, see `.github/workflows/on_release.yml` (last action - "Deploy Grafolean to Docker Hub").

Let me know if you would like to give it a try.

**Reply by azzajess** [reply]
Thanks for looking into it, Appreciate it. I will do my best to look into to see if I can get anywhere, but cant promise anything :P
Seems a hell of a lot easier to just wait until a pi 0 2 comes in stock and get one of those.
Thanks again.
